### PR TITLE
Enhance User Profile Page

### PR DESCRIPTION
**Pull request by NipuniBhagya**

### CODE REVIEW

File: apps/user-portal/src/locales/en/views.json [hunk 1]
```diff
@@ -112,7 +112,9 @@
     "userProfile": {
         "title": "Profile",
         "subTitle": "Manage information about you, your sub profiles and your account in general.",
+        "personalInfoTitle": "Personal Information",
         "inputFields": {
+            "name": "Full Name",
```
Comment (jeradrutnam) [inline apps/user-portal/src/locales/en/views.json]: Let's make this "name" instead

File: apps/user-portal/src/models/profile.ts [hunk 2]
```diff
@@ -29,9 +29,10 @@ export interface BasicProfileInterface {
     roles: string[];
     proUrl: string;
     isSecurity: boolean;
-    firstNameEdit: boolean;
-    lastNameEdit: boolean;
+    nameEdit: boolean;
+    mobile: string;
     emailEdit: boolean;
+    pInfoEdit: boolean;
```
Comment (jeradrutnam) [inline apps/user-portal/src/models/profile.ts]: Is this "personalInfoEdit" ?

File: apps/user-portal/src/pages/user-profile.tsx [hunk 3]
```diff
@@ -18,16 +18,17 @@
 
 import * as React from "react";
 import { withTranslation, WithTranslation } from "react-i18next";
-import { Button, Container, Divider, Form, Grid, Header, Icon, Message, Segment, Transition } from "semantic-ui-react";
+import { Button, Container, Divider, Form, Grid, Header, Icon, Segment, Transition } from "semantic-ui-react";
 import { getProfileInfo, updateProfileInfo } from "../actions";
-import { NotificationComponent, UserImagePlaceHolder } from "../components";
+import { NotificationComponent, UserImagePlaceHolder} from "../components";
```
Comment (jeradrutnam) [inline apps/user-portal/src/pages/user-profile.tsx]: Missing space before `}`

File: apps/user-portal/src/pages/user-profile.tsx [hunk 4]
```diff
@@ -18,16 +18,17 @@
...
 import { InnerPageLayout } from "../layouts";
 import { createEmptyProfile } from "../models/profile";
 
 /**
  * Component Props types
  */
-interface IComponentProps extends WithTranslation { }
+interface IComponentProps extends WithTranslation {
```
Comment (jeradrutnam) [inline apps/user-portal/src/pages/user-profile.tsx]: Use `WithTranslation` directly instead. And avoid `I` prefixing for interfaces

File: apps/user-portal/src/pages/user-profile.tsx [hunk 5]
```diff
@@ -154,91 +179,140 @@ class UserProfilePageComponent extends React.Component<IComponentProps, any> {
...
+                                            </Form.Field>
+                                        </Grid.Column>
+                                    </Grid.Row>
+                                </Grid>
+                                <Divider hidden/>
+                                <Button id="pInfo" primary onClick={this.handleSave}>
                                     {t("common:save")}
                                 </Button>
-                                <Button id="emailEdit" onClick={this.handleCancel}>
+                                <Button id="pInfoEdit" secondary onClick={this.handleCancel}>
```
Comment (jeradrutnam) [inline apps/user-portal/src/pages/user-profile.tsx]: Why do we need a `id` here?

File: apps/user-portal/src/pages/user-profile.tsx [hunk 6]
```diff
@@ -263,40 +337,30 @@ class UserProfilePageComponent extends React.Component<IComponentProps, any> {
...
-                                    {this.state.phoneNumbers.map((mobile) => {
-                                        return (<div>{mobile.value}</div>);
-                                    })}
-                                </Grid.Column>
-                            </Grid.Row>
-                        </Grid>
+                        <Header dividing={true} as="h3">{t("views:userProfile.personalInfoTitle")}
+                            <Button basic id="pInfoEdit" onClick={this.handleEdit}
+                                    name="pencil alternate" size="small"
+                                    style={{marginLeft: "10px", padding: "6px"}}>Update</Button>
```
Comment (jeradrutnam) [inline apps/user-portal/src/pages/user-profile.tsx]: Avoid inline styling. This style will reuse in other places as well

File: apps/user-portal/src/pages/user-profile.tsx [hunk 7]
```diff
@@ -308,11 +372,16 @@ class UserProfilePageComponent extends React.Component<IComponentProps, any> {
...
+        let mNumber = "";
+        response.phoneNumbers.map((mobile) => {
+            mNumber = mobile.value;
+        });
         this.setState({
             displayName: response.displayName,
             email: response.emails[0],
             emails: response.emails,
             lastName: response.lastName,
+            mobile: mNumber,
```
Comment (jeradrutnam) [inline apps/user-portal/src/pages/user-profile.tsx]: Use full qualified names instead

File: apps/user-portal/src/pages/user-profile.tsx (hunk 5 above)
Comment (NipuniBhagya) [reply apps/user-portal/src/pages/user-profile.tsx]: This id is passed into the event handler of the cancel button.

**Pull request merged**